spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/SparkReadConf.java

```diff
@@ -255,6 +255,22 @@ public Long endTimestamp() {
     return confParser.longConf().option(SparkReadOptions.END_TIMESTAMP).parseOptional();
   }
 
+  public Integer maxFilesPerMicroBatch() {
```

Comment:
Why return `Integer` if the value can't be null?

Reply:
Agree with you, ideally it's not required, I think might have just picked up from previous conf's like below : 
https://github.com/apache/iceberg/blob/e2c7e77310329b2113d4122eda4bd0d46e1d252d/spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/SparkReadConf.java#L242-L248


Let me create a pr to clean up the conf's introduced above as well as existing conf's to just return primitive types, if it's not optional.

Reply:
Added a pr for the same https://github.com/apache/iceberg/pull/7429/files

Reply:
I wonder if this is the right approach. Using boxed reference for many configs simply avoids the need to use extreme values like int/long min/max. I feel it makes more sense to keep that behavior and remove the usage of `Long.MIN_VALUE` here? Any thoughts?

Reply:
I see, after reading the linked PR, looks like we are using those int/long min/max all the time. In that case it's probably fine to change them to unboxed primitives since it's established pattern, and the value is always set. I approved that PR.